site/content/entry/yuzu-progress-report-jul-2023/index.md

```diff
@@ -0,0 +1,95 @@
+
+So, what's new, you ask?
+The brains behind this feature's implementation and our trusty slice of toast, [toastUnlimited](https://github.com/lat9nq/), started working towards a future where yuzu could essentially download a game-specific configuration file, that was curated by the community, and then instantly apply those settings.
+Thus users obtaining the optimal settings for any game without having to tinker with the several configuration options we have today.
+
+Sounds exciting, right? Heck yeah!
+While toast was finalizing his designs, he quickly ran into a road-block of his own design that is our current per-game configuration system.
+The existing system's design made it difficult to programmatically define and override each setting in multiple places across yuzu's backend and frontend systems.
+
+toast decided to take the challenge head-on and {{< gh-hovercard "10839" "rewrote the entire settings backend" >}} to improve it while making it programmatically easier to define and implement any setting.
+Thanks to these changes, developers now only need to write out the UI text for a setting and the system will do the rest (reading, writing, representing in the UI, resetting global state).
+
+Note: These changes are currently for developers only with no end-user impact, but do keep your eyes open for follow-up changes that will build on top of this foundation.
+
+
+# Graphics Changes
+
+It's July and We have another new AMD GPU driver with yet another extension broken.
+If you recall from our June progress report, we reported that latest AMD drivers had broken a Vulkan extension - `extendedDynamicState3ColorBlendEquation`, and we had to temporarily disable usage of this extension on AMD driver version `23.5.2` and above.
+
+Fast-forward to July, and it's still broken, kind of.
+Giving credit where due, AMD ***did*** fix this issue with driver version `23.7.2` but only for `RDNA2`, with at least `Polaris` confirmed to be broken.
+In light of this, We {{< gh-hovercard "11182" "reverted" >}} {{< gh-hovercard "11163" "our revert" >}} to {{< gh-hovercard "10946" "our original change from June" >}} and have {{<gh-hovercard "11204" "currently disabled usage of this extension on all AMD drivers" >}}, until AMD fixes this.
+
+[Maide](https://github.com/Kelebek1) continued his efforts on improving performance through optimizations and {{< gh-hovercard "10996" "his latest changes further optimize" >}} {{< gh-hovercard "10457" "one of his earlier optimizations" >}}.
+While the original changes aimed to reduce guest memory (Switch) allocations while copying data from guest to host (User's PC), the latest changes remove data copying entirely thereby resulting in nearly 10% performance boost in parts of "Super Mario Odyssey" like the metro area and "Xenoblade Chronicles: Definite Edition" reporting similar gains.
```

Comment:
While the original changes aimed to reduce guest memory (Switch) allocations while copying data from guest to host (User's PC), the latest changes remove data copying entirely thereby resulting in nearly 10% performance boost in parts of `SUPER MARIO ODYSSEY` like the Metro Kingdom area, and `Xenoblade Chronicles: Definite Edition` reporting similar gains.